spec/requests/vendor_api/post_make_an_offer_spec.rb

```diff
@@ -121,12 +121,12 @@
 
       expect(response).to have_http_status(422)
       expect(parsed_response).to be_valid_against_openapi_schema('UnprocessableEntityResponse')
-      expect(error_response['message']).to match 'Offered course provider ABC does not exist'
+      expect(error_response['message']).to match 'Cannot find an appropriate course option for these codes'
     end
   end
 
   describe 'offering for application with a decision' do
-    it 'allows amending of existing offers' do
+    it 'allows amending of conditions for existing offers' do
```

Comment:
This has started failing now (with 'cannot find application with ID...'), even though the new scenario which uses the exact same methodology at L175 passes. Any ideas?

Reply:
Turns out the problem isn't the test failing, it's the expectation. For existing offers we now call `ChangeOffer` (which was the main goal of this work!), and this service checks if `course_option_id` is being changed before making any changes, to avoid the API resetting DBD at will.

I think it's fine as it is. Currently we do not allow changes to conditions for an existing offer in the UI. If we start providing this functionality, we can revisit such modifications through the API and discuss what should happen with DBD etc.